### PR TITLE
Introduce feature flagging for time crate

### PR DESCRIPTION
**Pull request by workingjubilee**
This complements the deprecation process that has already begun by allowing us to make this crate an optional dependency in a future version of PGX.

### CODE REVIEW

**Comment by workingjubilee** [comment]
For reference, this feature flag doesn't break any user's API as the flagged impls were introduced during `0.5.0-beta.0`

File: README.md [hunk 1]
```diff
@@ -227,7 +227,19 @@ There's a few things on our immediate TODO list
 will be built into the `cargo-pgx` subcommand and make use of https://github.com/zombodb/postgres-parser.
  - More examples -- especially around memory management and the various derive macros `#[derive(PostgresType/Enum)]`
 
-## Experimental Features
+
+## Feature Flags
+PGX has optional feature flags for Rust code that do not involve configuring the version of Postgres used,
+but rather extend additional support for other kinds of Rust code. These are not included by default.
+
+### "time-crate": interop with the `time` crate
```
Comment (johnrballard) [inline README.md:235]: This name works for me.  Good job finding a name.

**Review by johnrballard — APPROVED** [review]
LGTM

File: pgx-tests/Cargo.toml [hunk 2]
```diff
@@ -35,7 +35,6 @@ rustdoc-args = ["--cfg", "docsrs"]
 owo-colors = "3.4.0"
 once_cell = "1.10.0"
 libc = "0.2.126"
-pgx = { path = "../pgx", default-features = false, version= "=0.5.0-beta.0" }
```
Comment (BradyBonnette) [inline pgx-tests/Cargo.toml:38]: 🤔

**Review by BradyBonnette — COMMENTED** [review]
Everything here looks great!

Will this impact builds/testing/CI that much? Or will it only affect how pgx is built for version 0.5.0-beta.0?

**Comment by workingjubilee** [comment]
> Everything here looks great!
> 
> Will this impact builds/testing/CI that much? Or will it only affect how pgx is built for version 0.5.0-beta.0?

I don't think it will have much of an impact in practice, since we enable it in pgx-tests.

**Pull request merged**